anchor/anchorparser.go

```diff
@@ -28,18 +28,20 @@ func ParseAnchors(protoAnchors []*yotiprotoattr.Anchor) []*Anchor {
 
 		for _, cert := range parsedCerts {
 			for _, ext := range cert.Extensions {
-				var ae anchorExtension
 				if sourceOID.Equal(ext.Id) {
 					anchorType = AnchorTypeSource
+
+					extension := unmarshalExtension(ext.Value)
+					if extension != "" {
+						extensions = append(extensions, extension)
```

Comment:
That is in essence what I'm saying.

It's a shame the documentation from the protobufs has been stripped, the documentation I've seen clearly states the first certificate in the chain will have the source/verifier extension, this means if all we're interested in is the source/verifier (i.e. not verifying the chain) then we only need to process/parse the first certificate.

In addition to that a certificate may have many extensions (e.g. key use), but the leaf certificate will only have one Yoti source or one Yoti verifier extension, it may have other non-Yoti extensions. Therefore to find the anchor type and the source/verifier value we just need to loop through the extensions until a Yoti OID is found.

This code currently works because there is only ever one source or one verifier extension in the chain. If it were to ever process a malformed anchor (i.e. multiple source/verifier extensions) then the values slice will have multiple entries, but the type will only be of the last extension processed. This isn't too much of an issue if the anchor was trying to express multiple verifications (type would be verification, with multiple values), but if there was a mixture then both the type and values would be meaningless.